crml/staking/src/rewards/mod.rs

```diff
@@ -146,6 +148,18 @@ decl_module! {
 			ensure_root(origin)?;
 			ForceFiscalEra::put(true);
 		}
+
+		fn on_initialize(now: T::BlockNumber) -> Weight {
+			// payouts are scheduled to happen on quantized block intervals such that only blocks where
+			//  `now` % `BlockPayoutInterval` == 0 will every have a payout. see `schedule_reward_payouts`
+			if (now % T::BlockPayoutInterval::get()).is_zero() {
+				if let Some((validator_stash, amount)) = ScheduledPayouts::<T>::take(now) {
+					return T:: ScheduledPayoutRunner::run_payout(&validator_stash, amount, Self::scheduled_payout_era())
```

Comment:
This is not a good load balancing when one validator has got many more nominators than another one.